### PR TITLE
Update git2 to 0.7.0

### PR DESCRIPTION
**Pull request by alexcrichton**
cc #5066

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -28,16 +28,16 @@ failure = "0.1.1"
 filetime = "0.1"
 flate2 = "1.0"
 fs2 = "0.4"
-git2 = "0.6.11"
-git2-curl = "0.7"
+git2 = "0.7.0"
```
Comment (matklad) [inline Cargo.toml]: Don't know if this is intentional or not, but looks like https://github.com/alexcrichton/git2-rs does not contain commit which actually bumps the versions?

Comment (alexcrichton) [reply Cargo.toml]: Oops forgot to do `git push`

**Comment by matklad** [comment]
Verified that it actually does work on a win7 virtualbox!

**Comment by matklad** [comment]
CI failure seems legit: https://ci.appveyor.com/project/rust-lang-libs/cargo/build/1.0.3722#L1458

**Pull request merged**